### ISSUE TITLE
Some comments about the code

### ISSUE THREAD

**Issue by LilySnow** [X-lab-3D/GradPose#11]

**Comment by LilySnow** [comment]
1. why do we need the option of use_numpy
2. rotate_all_pool(): a. the name is not very straight-forward about what it does, b. this function only runs on cpu
3. please add code coverage as DeepRank does

**Comment by DTRademaker** [comment]
1)  Torch combined with pool multithreading gives issues, with numpy it does not. We don't use gradients during converting of pdbs so numpy works the same as torch given the quaternions. So we use torch for optimization and Numpy for converting pdbs.  Said that, LINE 367 specifically sets use_numpy to true , so its place in the init seems odd indeed.. @KevinVG207 is it necesarry to have the use_numpy in the init?
2) ok, @KevinVG207  better name idea?
3) no idea what that is, kevin? @KevinVG207 
4) yes, 427 is a different optimizer with different learning rate than 59
5) loss = mse not rmsd
6) Never heard of that class before. Also, less classes and lbraries keeps code readable and easy to change. 
7) Same numbering is the assumption right? We have no idea if numbering is consistent between pdbs or not. Or do you mean to give a warning if some have deletions, which we can detect?
8) @KevinVG207 the quat branch is your experiment :)

**Comment by KevinVG207** [comment]
1. I have removed the unneeded use_numpy from the init of the Rotator class.
2. I have renamed the pool functions to include "multiprocessing" instead. The new function is rotate_all_with_multiprocessing()
3. I tried adding coveralls to the repository but I think there are some issues related to multiprocessing and the way I run the package as a subprocess during testing, so coverage is lower than it should be.
8. I will look at that soon.

**Comment by DTRademaker** [comment]
1. Where? both the main and development branches still have them
2. see comment above, forgot to upload to github?
3. Nice work! I placed it on top instead of on the bottom :)

**Comment by KevinVG207** [comment]
@DTRademaker 
My bad, I did indeed forget to push the changes for 1. and 2. The changes are now in both branches.

**Comment by DTRademaker** [comment]
Thank you for your feedback!

Regarding point 1, we have found that using both numpy and torch in the appropriate places can provide the most efficient processing of all files. Both libraries have their benefits and limitations, and we have carefully optimized the code to take advantage of their strengths.

For point 2, you are correct that GPU acceleration can be beneficial during training, but its usefulness during the rotating of PDBs can be limited. GPU acceleration does not take advantage of batching in this step and can result in higher latency due to data transfers between CPU and GPU memory. These factors slow down the processing of PDB files these cases, and the benefits of GPU acceleration may not be as significant for this step as they are for training.
